src/pytest_plugins/test_filler/test_filler.py

```diff
@@ -150,6 +151,29 @@ def pytest_addoption(parser):
             "The --html flag can be used to specify a different path."
         ),
     )
+    test_group.addoption(
+        "--strict-alloc-mode",
+        action="store_true",
+        dest="strict_alloc",
+        default=False,
+        help=("[DEBUG ONLY] Disallows deploying a contract in a predefined address."),
+    )
+    test_group.addoption(
+        "--test-contract-start-address",
+        action="store",
+        dest="test_contract_start_address",
+        default=None,
+        type=str,
+        help="The starting address from which tests will deploy contracts.",
+    )
+    test_group.addoption(
+        "--test-contract-address-increments",
+        action="store",
+        dest="test_contract_address_increments",
+        default=None,
+        type=str,
+        help="The address increment value to each deployed contract by a test.",
+    )
```

Comment:
I believe we could also add a check that prevents the starting address from being set to the existing precompile address space. The naive version could simply block the latest fork, but we could additionally do this on per fork basis. We'd use the `precompiles()` class function within `forks.py` determine this. 

Again this could be added to a seperate issue for adding later etc.

Reply:
I prefer the original, more verbose command-line options. But we can add both if you like, you can add synonyms, e.g.,
https://github.com/ethereum/execution-spec-tests/blob/2df7dea50c506cc1f5eb6c57c9a54dd7d660a8cf/src/pytest_plugins/test_filler/test_filler.py#L155-L158

Reply:
Added both.